client/src/__tests__/login.test.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { render, fireEvent, waitFor } from '@testing-library/react'
+import { MockedProvider } from '@apollo/client/testing'
+import { MemoryRouter, Route, Routes } from 'react-router-dom'
+import Login, { SIGN_IN_MUTATION } from '../pages/Login'
+
+// Mock the useNavigate hook
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: jest.fn(),
+}))
+
+const mockNavigate = jest.fn()
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockNavigate,
+}))
```

Comment:
Since mocking this hook will be utilized multiple times in other tests, we can abstract this functionality into another file.
Create a folder within `__tests__` known as `utils`. Add a file known as `navigation.ts`  and add the mocks.